File: _1550_three_consecutive_odds/main.cpp
class Solution {
public:
    bool threeConsecutiveOdds(vector<int>& arr) {
        unsigned char numberOfOdds = 0;
        for (const int& number : arr) {
            if (number % 2 == 0) {
                numberOfOdds = 0;
                continue;
            }

            numberOfOdds += 1;
            if (numberOfOdds == 3) {
                return true;
            }
        }

        return false;
    }
};

File: _1550_three_consecutive_odds/main.rs
impl Solution {
    pub fn three_consecutive_odds(arr: Vec<i32>) -> bool {
        arr.windows(3).any(|n| n[0] % 2 == 1 && n[1] % 2 == 1 && n[2] % 2 == 1)
    }
}